### PR TITLE
Adding Bynder browse media UI Extension

### PR DESCRIPTION
**Pull request by floelhoeffel**

### CODE REVIEW

File: samples/bynder/extension.html [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+            renderIds(api.field.getValue())
+
+            var isOpen = false
+
+            document.addEventListener('BynderAddMedia', function (e) {
+                isOpen = false
+                // The selected assets go in the event detail property
+                var selectedAssets = e.detail
+                var assetIds = []
```
Comment (jelz) [inline samples/bynder/extension.html]: `map` should not side effect. Should be:

```js
var assetIds = e.detail.map(function (asset) {
  return asset.id
})
```

File: samples/bynder/extension.html [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                selectedAssets.map(function (asset) {
+                    assetIds.push(asset.id)
+                })
+                renderIds(assetIds)
+                api.field.setValue(assetIds)
+                api.window.updateHeight()
+            })
+
+
+            document.getElementById('bynder-compactview').addEventListener('click', function (e) {
```
Comment (jelz) [inline samples/bynder/extension.html]: you don't use `e`, skip it

File: samples/bynder/extension.html [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                if (isOpen === false) {
```
Comment (jelz) [inline samples/bynder/extension.html]: `if (!isOpen) {`

File: samples/bynder/extension.html [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+                    isOpen = true
+                    api.window.updateHeight(500)
+                }
+            })
+
+            function renderIds(ids) {
+                if (!ids || ids.length === 0) {
```
Comment (jelz) [inline samples/bynder/extension.html]: More defensive: `!Array.isArray(ids) || ids.length < 1`

File: samples/bynder/extension.html [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+                    assetsListsContainer.style.display = 'none'
+                    api.window.updateHeight()
+                    return
+                }
+                assetsListsContainer.style.display = 'block'
```
Comment (jelz) [inline samples/bynder/extension.html]: Do it after HTML of the list is fully generated (your `innerHTML` manipulation)

File: samples/bynder/extension.html [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+                        item,
+                        '</li>'
+                    ].join('')
+                })
+                assetsList.innerHTML += '</ul>'
+                api.window.updateHeight()
+            }
+        })
+    </script>
+</body>
```
Comment (jelz) [inline samples/bynder/extension.html]: Add a new line after

File: samples/bynder/extension.html [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    <div id="bynder-compactview" data-fullScreen="true" data-assetTypes="image,video" data-autoload="false" data-button="Load media from bynder.com"
+        data-collections="true" data-folder="bynder-compactview" data-fullScreen="false" data-header="true"
+        data-language="en_US" data-mode="multi" data-zindex="300" data-defaultEnvironment="https://contentful.getbynder.com">
+    </div>
+    <div id="selection">
+        <p>Selected media:</p>
+        <div id="importedAssets"></div>
+    </div>
+
+    <script src="https://d8ejoa1fys2rk.cloudfront.net/modules/compactview/includes/js/client-1.1.0.min.js"></script>
```
[the remark this answers is not in the document]
Comment (jelz) [reply samples/bynder/extension.html]: In this case move SDK `<script>` here

File: samples/bynder/extension.html (hunk 1 above)
Comment (jelz) [reply samples/bynder/extension.html]: This is not addressed

File: samples/bynder/extension.html (hunk 2 above)
Comment (jelz) [reply samples/bynder/extension.html]: This is not addressed

File: samples/bynder/extension.html (hunk 7 above)
[the remark this answers is not in the document]
Comment (floelhoeffel) [reply samples/bynder/extension.html]: done

File: samples/bynder/extension.html (hunk 2 above)
Comment (floelhoeffel) [reply samples/bynder/extension.html]: done

File: samples/bynder/extension.html [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+            var assetsListsContainer = document.getElementById('selection')
+            var assetsList = document.getElementById('importedAssets')
+
+            renderIds(api.field.getValue())
+
+            var isOpen = false
+
+            document.addEventListener('BynderAddMedia', function () {
+                isOpen = false
+                var assetIds = e.detail.map(function (asset) {
```
Comment (jelz) [inline samples/bynder/extension.html]: `e` is used here

File: samples/bynder/extension.html [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+
+            function renderIds(ids) {
+                if (!Array.isArray(ids) || ids.length < 1) {
+                    assetsListsContainer.style.display = 'none'
+                    api.window.updateHeight()
+                    return
+                }
+                assetsList.innerHTML = '<ul style="clear:both">'
+                ids.map(function (item) {
+                    assetsList.innerHTML += [
```
Comment (jelz) [inline samples/bynder/extension.html]: I'd say this:

```js
assetsList.innerHTML += ('<li>' + item + '</li>');
```

is both shorter and more readable

File: samples/bynder/README.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+## TODOs
+
+- the compact view is not showing previously selected media and will always reset the selection of a previous session
+
+## Usage
+
+For local development, start a webserver:
+
+```bash
+❯python -m SimpleHTTPServer
```
Comment (jelz) [inline samples/bynder/README.md]: maybe drop the `>` so it can be copy/pasted w/o confusion

**Pull request merged**